applications/integrationtests/geometryConcrete/CMakeLists.txt

```diff
@@ -18,27 +18,17 @@ if(ENABLE_PYTHON)
         set_property(TEST integration::geometryConcrete::${test}.py
             PROPERTY ENVIRONMENT "PYTHONPATH=${CMAKE_BINARY_DIR}/python")
     endforeach()
-
-    if(GMSH_PATH)
-        add_test(integration::geometryConcrete::GmshExport.py
-            ${CMAKE_CURRENT_SOURCE_DIR}/GmshExport.py ${GMSH_PATH})
-        set_property(TEST integration::geometryConcrete::GmshExport.py
-            PROPERTY ENVIRONMENT "PYTHONPATH=${CMAKE_BINARY_DIR}/python")
-    else()
-        message(STATUS
-            "Gmsh not found. Skipping integration test GmshExport.py")
-    endif()
 endif()
 
 if(GMSH_PATH)
-    add_executable(GeometryConcreteAPI EXCLUDE_FROM_ALL GeometryConcreteAPI.cpp)
-    target_link_libraries(GeometryConcreteAPI
-        Math Base GeometryConcrete Boost::unit_test_framework)
-    target_include_directories(GeometryConcreteAPI
-        PRIVATE ${CMAKE_SOURCE_DIR}/test/tools)
-    add_test(integration::geometryConcrete::GeometryConcreteAPI.cpp
-        GeometryConcreteAPI ${GMSH_PATH})
-    append_to_tests(GeometryConcreteAPI)
+    add_integrationtest(GeometryConcreteAPI)
+    target_link_libraries(GeometryConcreteAPI GeometryConcrete)
+
+    add_executable(GmshWriter EXCLUDE_FROM_ALL GmshWriter.cpp)
```

Comment:
You don't need the `EXCLUDE_FROM_ALL` here, because everything in `integrationtests/` is excluded. Also, have you considered adding the additional option you pass to the test to the `add_integrationtest` macro, which you could then reuse?

Reply:
good point!